src/core.c/CompUnit/RepositoryRegistry.pm6

```diff
@@ -332,15 +370,36 @@ class CompUnit::RepositoryRegistry {
             exit 1;
         }
 
-        my $meta = @metas.sort(*.<ver>).sort(*.<api>).reverse.head;
-        my $bin  = $meta<source>;
+        my $distro = @distros.sort(*.meta<ver>).sort(*.meta<api>).reverse.head;
+        my $bin  = $distro.meta<source>;
+        my $*DISTRIBUTION := $distro;
         require "$bin";
     }
 
     method head() { # mostly usefull for access from NQP
         $*REPO
     }
 
+    method NEED(Str:D $module_name, %opts) is implementation-detail {
```

Comment:
This is another method that seems a bit out of place. For example why is there no `LOAD` / `require` logic? This seems to be further splitting up the loading logic between `need` and `load`